GitUI/BranchTreePanel/LocalBranchMenuItemsBehavior.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using GitUI.BranchTreePanel.Interfaces;
+
+namespace GitUI.BranchTreePanel
+{
+    internal class LocalBranchMenuItemGenerator<TNode> : MenuItemsGenerator<TNode> where TNode : class, INode
+    {
+        public LocalBranchMenuItemGenerator(IMenuItemFactory menuItemFactory) : base(menuItemFactory)
+        {
+        }
+
+        /// <summary>
+        /// Filter menu entries that depend on the branch being active or not
+        /// </summary>
+        public IEnumerable<ToolStripItemWithKey> GetInactiveBranchItems()
+        {
+            MenuItemKey[] keysToRetrieve =
+                new[]
+                {
+                    MenuItemKey.GitRefCheckout,
+                    MenuItemKey.GitRefMerge,
+                    MenuItemKey.GitRefMerge,
+                    MenuItemKey.GitRefRebase,
+                    MenuItemKey.GitRefReset,
+                    MenuItemKey.GitRefActionsSeparator,
+                    MenuItemKey.Delete
+                };
+
+            return this.Where(t => keysToRetrieve.Contains(t.Key));
+        }
+
+        protected override string GetTooltip(MenuItemKey key)
+        {
+            switch (key)
+            {
+                case MenuItemKey.Rename:
+                    return "Rename this branch";
```

Comment:
The returned value is a translation key, not a translated string... it will be handled be CreateMenuItem and transformed in a new TranslationString. The final item will have the text assigned through a TranslationString. Isn't this correct? ❗️

Reply:
```
result.Text = new TranslationString(captionKey).Text;
result.ToolTipText = new TranslationString(toolTipKey).Text;
```

Reply:
@RussKie @amaiorano I understand now how the translation system works. I currently have work in progress in order to use it correctly